dch/public/app/components/watch-form/watch-form.js

```diff
@@ -0,0 +1,205 @@
+(function() {
+
+	'use strict';
+
+  /**
+   * @desc - Directive used to create the watch form query. Posts
+   *         user selected information to the DCH server.
+   *
+   * @example <watch-form></watch-form>
+   */
+   angular
+    .module('dcgui.components')
+    .directive('watchForm', watchForm)
+    // This filter is temporary. Will later use the API to directly
+    // retrieve the WatchIDs. Right now it simply removes duplicates
+    // of WatchIDs when populating the list of WatchIDs.
+    .filter('unique', function(){
+    	return function(collection, keyname) {
+    		var output = [],
+    		keys = [];
+
+    		angular.forEach(collection, function(item) {
+    			var key = item[keyname];
+    			if(keys.indexOf(key) === -1) {
+    				keys.push(key);
+    				output.push(item);
+    		}
+    	});
+    	return output;
+    	}
+    });
+
+    function watchForm() {
+
+    	var directive = {
+    		scope: {
+    			watches: '='
+    		},
+    		templateUrl: '/app/components/watch-form/watch-form.html',
+    		controller: WatchFormController,
+    		controllerAs: 'watchFormCtrl',
+    		restrict: 'E',
+    		bindToController: true
+    	};
+
+    	return directive;
+
+    }
+
+    WatchFormController.$inject = ['$http', '$scope'];
+
+    function WatchFormController($http, $scope) {
+
+        $scope.watchSelected = false;
+        $scope.timeSelected = false;
+        $scope.times = [];
+
```

Comment:
We talked about $scope a bit before. It's generally advised that they be avoided unless they are needed (and there are events when they're needed) but in this case, you can access these values by using the ``` vm.watchSelected = false; ``` as an example. It may seem verbose but it keeps the code a lot more readable in the long run (especially in the HTML files that this controller binds to).